src/linux.rs

```diff
@@ -57,11 +57,19 @@ pub fn enumerate_platform(vid: Option<u16>, pid: Option<u16>) -> Vec<UsbDevice>
                     .map(|s| s.to_string());
             }
 
+            let serial_number = device
+                .property_value("ID_SERIAL_SHORT")
+                .ok_or(ParseError)?
+                .to_str()
+                .ok_or(ParseError)?
+                .to_string();
```

Comment:
If there is no `ID_SERIAL_SHORT` this will result in an error rather than a device with `serial_number` being `None`.

I think this should copy the same logic as `ID_MODEL_FROM_DATABASE`, ie:

```rust
            let mut description = device
                .property_value("ID_MODEL_FROM_DATABASE")
                .and_then(|s| s.to_str())
                .map(|s| s.to_string());
```